db/l2db/views.go

```diff
@@ -38,64 +38,105 @@ type PoolTxAPI struct {
-	TokenName        string            `meddler:"name"`
-	TokenSymbol      string            `meddler:"symbol"`
-	TokenDecimals    uint64            `meddler:"decimals"`
-	TokenUSD         *float64          `meddler:"usd"`
-	TokenUSDUpdate   *time.Time        `meddler:"usd_update"`
+	BatchNum             *common.BatchNum      `meddler:"batch_num"`
+	Timestamp            time.Time             `meddler:"timestamp,utctime"`
+	TotalItems           uint64                `meddler:"total_items"`
+	TokenID              common.TokenID        `meddler:"token_id"`
+	TokenItemID          uint64                `meddler:"token_item_id"`
+	TokenEthBlockNum     int64                 `meddler:"eth_block_num"`
+	TokenEthAddr         ethCommon.Address     `meddler:"eth_addr"`
+	TokenName            string                `meddler:"name"`
+	TokenSymbol          string                `meddler:"symbol"`
+	TokenDecimals        uint64                `meddler:"decimals"`
+	TokenUSD             *float64              `meddler:"usd"`
+	TokenUSDUpdate       *time.Time            `meddler:"usd_update"`
 }
 
 // MarshalJSON is used to neast some of the fields of PoolTxAPI
 // without the need of auxiliar structs
 func (tx PoolTxAPI) MarshalJSON() ([]byte, error) {
-	return json.Marshal(map[string]interface{}{
-		"itemId":                      tx.ItemID,
-		"id":                          tx.TxID,
-		"type":                        tx.Type,
-		"fromAccountIndex":            tx.FromIdx,
-		"fromHezEthereumAddress":      tx.EffectiveFromEthAddr,
-		"fromBJJ":                     tx.EffectiveFromBJJ,
-		"toAccountIndex":              tx.ToIdx,
-		"toHezEthereumAddress":        tx.EffectiveToEthAddr,
-		"toBJJ":                       tx.EffectiveToBJJ,
-		"amount":                      tx.Amount,
-		"fee":                         tx.Fee,
-		"nonce":                       tx.Nonce,
-		"state":                       tx.State,
-		"maxNumBatch":                 tx.MaxNumBatch,
-		"info":                        tx.Info,
-		"errorCode":                   tx.ErrorCode,
-		"errorType":                   tx.ErrorType,
-		"signature":                   tx.Signature,
-		"timestamp":                   tx.Timestamp,
-		"requestFromAccountIndex":     tx.RqFromIdx,
-		"requestToAccountIndex":       tx.RqToIdx,
-		"requestToHezEthereumAddress": tx.RqToEthAddr,
-		"requestToBJJ":                tx.RqToBJJ,
-		"requestTokenId":              tx.RqTokenID,
-		"requestAmount":               tx.RqAmount,
-		"requestFee":                  tx.RqFee,
-		"requestNonce":                tx.RqNonce,
-		"token": map[string]interface{}{
-			"id":               tx.TokenID,
-			"itemId":           tx.TokenItemID,
-			"ethereumBlockNum": tx.TokenEthBlockNum,
-			"ethereumAddress":  tx.TokenEthAddr,
-			"name":             tx.TokenName,
-			"symbol":           tx.TokenSymbol,
-			"decimals":         tx.TokenDecimals,
-			"USD":              tx.TokenUSD,
-			"fiatUpdate":       tx.TokenUSDUpdate,
+	type jsonToken struct {
```

Comment:
The is struct is perfect. However, it needs to be defined in common package and public in order to be used for external lib/binaries such as SDK. Could you move both to other package or at least define them as public in this package?

Reply:
If we're moving this into common, I'd like to suggest to do something like:

```go
// common/pooll2tx.go
package common

type PoolL2TxAPI struct // similar to l2db.PoolL2TxAPI BUT with neasted token struct instead of flat
func (tx PoolTxAPI) MarshalJSON() ([]byte, error)
func (tx *PoolTxAPI) UnmarshalJSON(data []byte) error


// db/l2db/views.go
package l2db

type poolTxAPIView struct // same as l2db.PoolTxAPI
func poolTxViewToAPI(tx poolTxAPIView) common.PoolL2TxAPI
``` 

Rationale: we're using a flat struct in order to be able to put the data from the SQL rows, and we were making the nesting of the sub-struct (token) at the JSON marshal level, however if this struct has to be used elsewhere, it's very ugly for it to  be flat. Therefore I think the best option is to have the good non-flat struct at `common`, and let the `l2db` deal with the limitations of the underlying DB (note that this is very specific to SQL, if one day we switch to a noSQL it's likely that this won't be a problem and we can return the struct with extra transformation).

Reply:
I think this struct should not be moved to other place. Also, already exist a similar struct there https://github.com/hermeznetwork/hermez-node/blob/f39996efbb9130749bc6476bd7fd257b9e20af10/db/historydb/views.go#L244. 

If we're think to do a move (refactor), we should do a really big effort to this, because on the project will there repeated struct's. I think the purpose of PR is done.

Reply:
I guess there is a misunderstanding here. The issue was initially opened to allow SDK and other libraries to reuse structs  returned in API responses. and avoid we need to rebuild the same struct in other repo. It was not intent a full refactor. That's why without make them public the PR does not achieve issue's requirement.